### PR TITLE
feat(site launch): recommending www domain 

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

During the user trial, user was confused on the format in which to add in the url  + which option to choose regarding the `www` vs `non-www` (this was already a power user of isomer who already had experience launching numerous sites before. 

## Solution

This PR addresses that pain point to make the flow clearer. Note that the general rule of thumb here is that 3rd level domains -> have www, 4h level -> no www
The exception here are schools with `<blah>.moe.edu.sg`, which we have www at the 4th level, and no www at the 5th level and above.

This PR also adds a bit of loose URL validation (commented in the PR). My hunch here is that strict url validation is not a requirement here, and thus I rather keep this loose for flexibility + avoid false negatives. 

https://github.com/isomerpages/isomercms-frontend/assets/42832651/0fbd93f5-9234-4fd2-b257-1603e7c10347

## Alternatives considered
Input sanitation. Opted for validation here due for a quicker solution + absolute clarity to the user and bring to attention the __**exact**__ url that is going to be launched. 

## Designer Notes 
@sehyunidaaa, as discussed now there will be 3 error states. The video was taken when this PR was created, please refer to [here](https://www.chromatic.com/build?appId=61fcc8e14c3af9003a102ac9&number=1928) for the corresponding storybooks! Need your help to accept baseline too! [Design review blocked by storybook, remove this once chromatic builds work again]

### CODE REVIEW

File: src/layouts/SiteLaunchPad/components/SiteLaunchInfoGathering.tsx [hunk 1]
```diff
@@ -60,50 +71,82 @@ export const SiteLaunchInfoCollectorBody = ({
...
+  const handleWwwQn = (response: string) => {
+    if (response === "") {
+      // disallow unselecting
+      return
+    }
+    setWwwSetting(response)
+  }
+
+  const inputDomain = watch("domain")
+  useEffect(() => {
```
Comment (kishore03109) [inline src/layouts/SiteLaunchPad/components/SiteLaunchInfoGathering.tsx:98]: useEffect is used here to prevent infinite re-renders

**Comment by kishore03109** [comment]
!run-chromatic

**Comment by sehyunidaaa** [comment]
Please review comments [on Chromatic](https://www.chromatic.com/test?appId=61fcc8e14c3af9003a102ac9&id=64d43eea175ed1920d3eaf3a)

File: src/layouts/SiteLaunchPad/components/SiteLaunchInfoGathering.tsx [hunk 2]
```diff
@@ -60,50 +71,86 @@ export const SiteLaunchInfoCollectorBody = ({
...
+  const [wwwSetting, setWwwSetting] = useState("")
+  const [displayWwwQn, setDisplayWwwQn] = useState(false)
+  const [isWwwRecommended, setIsWwwRecommended] = useState(true)
+
+  const handleDomainNatureQn = (response: string) => {
+    if (response === "") {
+      // keep the state as is, disallow unselecting
+      return
+    }
+    setIsWwwRecommended(recommendWwwValue(watch("domain")))
```
Comment (alexanderleegs) [inline src/layouts/SiteLaunchPad/components/SiteLaunchInfoGathering.tsx:84]: `setIsWwwRecommended` is being done in the useeffect right, is this still required here?

**Review by alexanderleegs — APPROVED** [review]
lgtm, minor nit

File: src/layouts/SiteLaunchPad/components/SiteLaunchInfoGathering.tsx (hunk 2 above)
Comment (kishore03109) [reply src/layouts/SiteLaunchPad/components/SiteLaunchInfoGathering.tsx]: thanks for the catch, fixed here 3f489afb5c0bb2674d10aba6aead6a8f2c36bcbd

**Comment by kishore03109** [comment]
!run-chromatic

**Pull request merged**